src/vanilla/Model/MethodTS.cs

```diff
@@ -774,7 +774,8 @@ public string GenerateResponseType(string emptyLine)
                         foreach (var property in ((CompositeTypeTS)ReturnType.Headers).Properties)
                         {
                             iface.DocumentationComment(property.Documentation);
-                            iface.Property(property.Name, property.ModelType.TSType(inModelsModule: true));
+                            // hard-coding response headers non-optional because Swagger itself can't specify whether or not they're required.
+                            iface.Property(property.Name, property.ModelType.TSType(inModelsModule: true), optional: false);
```

Comment:
will want to do this when generating headers model interfaces as well and will maybe solve that issue at the same time